system/Helpers/Array/ArrayHelper.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Helpers\Array;
+
+/**
+ * @see \CodeIgniter\Helpers\Array\ArrayHelperRecursiveDiffTest
+ */
+final class ArrayHelper
```

Comment:
I'm not a huge fan of making this type of class `final`, since there might be some use cases that can benefit from extending this class. However, it's not a deal breaker for me.

Reply:
If you need it really, I will remove `final`.
But we can remove it anytime when request comes.

Reply:
I don't feel like we need it here, but I may be the only one, so that's cool. And you're right, we can always remove it later.

Reply:
Let's wait for others opinions.